### PR TITLE
Ft authentication token

### PR DESCRIPTION
**Pull request by JayKay24**
**What does this PR do?**
Addition of authentication for HTTP requests

**Description of the task to be completed**
Have the following endpoints working
* POST api/v1/auth/register/
* POST api/v1/auth/login/
* GET api/v1/bucketlists/
* POST api/v1/bucketlists/
* PATCH api/v1/bucketlists/id
* DELETE api/v1/bucketlists/id
* POST api/v1/bucketlistitems/
* GET api/v1/bucketlistitems/
* PATCH api/v1/bucketlistitems/id
* DELETE api/v1/bucketlistitems/id

**How can it be manually tested?**
nose2 -v --with-coverage

### CODE REVIEW

File: tests/test_views.py [hunk 1]
```diff
@@ -22,19 +26,138 @@ def tearDown(self):
...
+        data = {"bkt_item_name": new_bucket_item_name_2}
+        patch_response = self.test_client.patch(
+            new_bucketl_item_url,
+            headers=self.get_authentication_headers(self.test_user_name, 
+                self.test_user_password),
+            data=json.dumps(data))
+        self.assertEqual(patch_response.status_code, status.HTTP_200_OK)
+
+
+
```
Comment (victoriaaoka) [inline tests/test_views.py]: Consider adding tests for edge-cases eg:
        - test for updating a bucketlist with the same information.
         - test for getting, updating and deleting bucketlists that do not exist.
         - test for getting, updating and deleting bucketlist-items that do not exist.
         - test for pagination and search functionalities.
         - test for user accessing bucketlists with expired or no token.
         - Test for username with numbers or special characters.

File: views.py [hunk 2]
```diff
@@ -172,24 +199,44 @@ def post(self):
...
             if Bucketlistitem.is_unique(id=id, bkt_item_name=bucketitem_bucketitem):
                 bucketlist_item.bkt_item_name = bucketitem_bucketitem
             else:
                 response = {"error": "A bucketlist item with the same name already exists"}
                 return response, status.HTTP_400_BAD_REQUEST
+        if 'bkt_name' in bucketlist_item_dict:
+            bucketlist = Bucketlist.query.filter_by(bkt_name=bucketlist_item_dict['bkt_name']).first()
+            if bucketlist is None:
+                response = {'error': 'No bucketlist by that name exists'}
+                return response, status.HTTP_400_BAD_REQUEST
```
Comment (victoriaaoka) [inline views.py]: How about using status_code 409?

**Review by victoriaaoka — COMMENTED** [review]
Nice code.
Consider adding comments and docstrings to the code.
